### PR TITLE
[DOCS] Add a roadmap section to the documentation.

### PR DESCRIPTION
**Pull request by clems4ever**
Adding the roadmap will likely help people figure out what are the next big
topics that might be missing for them to take the leap and use Authelia.
Maybe some users are also waiting for a feature to unlock some use cases.

### CODE REVIEW

File: docs/roadmap.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
+---
+layout: default
+title: Roadmap
+nav_order: 9
+---
+
+# Roadmap
+
+We are currently 3 globally distributed developers working actively on improving Authelia on our spare time and we define
```
Comment (nightah) [inline docs/roadmap.md:9]:
```suggestion
Currently the team consists of 3 globally distributed developers working actively on improving Authelia in our spare time and we define
```

File: docs/roadmap.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+our priorities based on a roadmap that we share here for transparency. We also try to balance it as much as possible with
```
Comment (nightah) [inline docs/roadmap.md:10]:
```suggestion
our priorities based on a roadmap that we share here for transparency. We also try to balance features and improves as much as possible with
```

File: docs/roadmap.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+the maintenance tasks we have to do to keep the backlog of open issues in a reasonable state.
```
Comment (nightah) [inline docs/roadmap.md:11]:
```suggestion
the maintenance tasks we have to perform to keep the backlog of open issues in a reasonable state.
```

File: docs/roadmap.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+If you're willing to contribute and help us move forward faster, get in touch with us on Matrix. We'll be glad to share
+ideas and plans with you.
+
+Here are the items in the roadmap:
```
Comment (nightah) [inline docs/roadmap.md:15]:
```suggestion
Below are the prioritised roadmap items:
```

File: docs/roadmap.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+1. [Authelia acts as an OpenID Connect Provider](https://github.com/authelia/authelia/issues/189). This is a high
+priority because currently the only way to pass authentication information back to the protected app is through the
+use of HTTP headers as described
+[here](https://docs.authelia.com/deployment/supported-proxies/#how-can-the-backend-be-aware-of-the-authenticated-users)
+but many apps do not support it while many of them support (or start to support) OpenID Connect or OAuth2 either
```
Comment (nightah) [inline docs/roadmap.md:21]:
```suggestion
however, many apps either do not support this method or are starting to move away from this in favour of OpenID Connect or OAuth2
```

File: docs/roadmap.md [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+internally or via plugins.
+
+2. [Administration interface](https://github.com/authelia/authelia/issues/974). This is useful in many cases to
+properly manage users and administrate activities like unbanning banned users. In the future we can even think of
+adding/removing users from there, request a password reset for a user or all users, request a 2FA enrollment,
+temporarily block users, etc...
+
+3. [User interface](https://github.com/authelia/authelia/issues/303). This will help the users manage their 2FA
+devices, reset their password, review their authentication activity.
+In the future we can imagine users will be able to customize their profile with an avatar, set their preferences
```
Comment (nightah) [inline docs/roadmap.md:31]:
```suggestion
In the future we envisage users will be able to customize their profile with an avatar, set their preferences
```

File: docs/roadmap.md [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+regarding 2FA and according to the global policy defined by administrators, etc...
+
+4. [Facilitate setup on Kubernetes](https://github.com/authelia/authelia/issues/974). There are mainly two objectives
+here. First, we need to provide the documentation required to setup Authelia on Kubernetes. Even though, some users
+already have it working and the feature is even tested in the project, there is a clear lack of documentation. The
+second item is to provide a Helm chart to ease even further the setup on Kubernetes.
```
Comment (nightah) [inline docs/roadmap.md:37]:
```suggestion
second item is to provide a Helm chart to streamline the setup on Kubernetes.
```

File: docs/roadmap.md (hunk 2 above)
Comment (james-d-elliott) [reply docs/roadmap.md:10]: Should be "features and improvements".
```suggestion
our priorities based on a roadmap that we share here for transparency. We also try to balance features and improvements as much as possible with
```

Comment (nightah) [reply docs/roadmap.md:10]: Yep good catch, thanks for that!

**Pull request merged**